utensor_cgen/ir/base.py

```diff
@@ -267,4 +273,5 @@ def __deepcopy__(self, memo):
     new_graph.ops_info = new_ops_info
     new_graph.topo_order = new_topo_order
     new_graph._backend = self._backend
+    new_graph.data_manager = self.data_manager
```

Comment:
You should make a copy of `data_manager` not just pass reference to new graph.
This violate the semantic of `deepcopy`.

Reply:
https://docs.python.org/3.6/library/copy.html

Reply:
if I use data_manager = attr.ib(default=None), should I have to deepcopy it in the new graph?

Reply:
Depends on how you'd like to use the data_manager.
If you don't want to share states among old and new graphs, you should make a deepcopy.